### PR TITLE
feat: Added docs using DocFx

### PR DESCRIPTION
**Pull request by sammychinedu2ky**
fixes https://github.com/oras-project/oras-dotnet/issues/11

### CODE REVIEW

File: Oras/Remote/Repository.cs
Comment (Wwwsylvia) [inline Oras/Remote/Repository.cs:1]: This change looks irrelevant to the purpose of this PR.

File: documentation/.gitignore [hunk 1]
```diff
@@ -0,0 +1,9 @@
+###############
+#    folder   #
+###############
+/**/DROP/
+/**/TEMP/
```
Comment (Wwwsylvia) [inline documentation/.gitignore:5]: Just curious what are DROP and TEMP?

File: documentation/articles/intro.md
Comment (Wwwsylvia) [inline documentation/articles/intro.md:1]: Is this currently in use?

File: documentation/index.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+ORAS works similarly to tools you may already be familiar with, such as docker. It allows you to push (upload) and pull (download) things to and from an OCI Registry, and also handles login (authentication) and token flow (authorization). What ORAS does differently is shift the focus from container images to other types of artifacts.
+
+ORAS is the de facto tool for working with OCI Artifacts. It treats media types as a critical piece of the puzzle. Container images are never assumed to be the artifact in question.
+
+By default, when pushing artifacts using ORAS, the `config.mediaType` field is set to `application/vnd.unknown.config.v1+json`.
+
+Authors of new OCI Artifacts are thus encouraged to define their own media types specific to their artifact, which their custom client(s) know how to operate on.
+
+If you wish to start publishing OCI Artifacts right away, take a look at the [ORAS CLI](https://oras.land/oras-cli). Developers who wish to provide their own user experience should use one of the [ORAS client libraries](https://oras.land/oras-client-libraries).
```
Comment (Wwwsylvia) [inline documentation/index.md:39]: The links are invalid now and can be hard to maintain in the future. To keep things simple, should we have a link pointing to the oras.land website and show just the API documentations here?

File: documentation/.gitignore (hunk 1 above)
Comment (sammychinedu2ky) [reply documentation/.gitignore:5]: They were scaffolded by the generator. Although not used

File: documentation/articles/intro.md
Comment (sammychinedu2ky) [reply documentation/articles/intro.md:1]: Nope

File: documentation/index.md (hunk 2 above)
Comment (sammychinedu2ky) [reply documentation/index.md:39]: Please Which links are invalid?

**Comment by Wwwsylvia** [comment]
We will need a GitHub action to upload the static site to GitHub Pages after this is merged. 😀

**Comment by sammychinedu2ky** [comment]
> We will need a GitHub action to upload the static site to GitHub Pages after this is merged. 😀

Sure would do that

File: Oras/Remote/Repository.cs
Comment (sammychinedu2ky) [reply Oras/Remote/Repository.cs:1]: It's for this https://github.com/oras-project/oras-dotnet/pull/28

File: documentation/.gitignore (hunk 1 above)
Comment (Wwwsylvia) [reply documentation/.gitignore:5]: Got it

File: documentation/articles/intro.md
Comment (Wwwsylvia) [reply documentation/articles/intro.md:1]: Should we remove it for now?

File: documentation/index.md (hunk 2 above)
Comment (Wwwsylvia) [reply documentation/index.md:39]: https://oras.land/oras-cli and https://oras.land/oras-client-libraries

Comment (sammychinedu2ky) [reply documentation/index.md:39]: Yh for maintenance, I would just point it to the home page

File: .github/workflows/deploy-to-github-pages.yml [hunk 3]
```diff
@@ -0,0 +1,52 @@
+name: Deploy DocFX Documentation
+'on':
```
[the remark this answers is not in the document]
Comment (sammychinedu2ky) [reply .github/workflows/deploy-to-github-pages.yml:2]: ooh that's a nit but still works 😄

File: .github/workflows/deploy-to-github-pages.yml [hunk 4]
```diff
@@ -0,0 +1,54 @@
+name: Deploy DocFX Documentation
+on:
+  push:
+    branches:
+      - main
+      - add-docs
```
Comment (Wwwsylvia) [inline .github/workflows/deploy-to-github-pages.yml:6]: We shouldn't include this in the yaml file.

File: documentation/index.md [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+Users seeking a generic registry client can benefit from the ORAS CLI, while developers can build their own clients on top of one of the ORAS client libraries.
+
+## What are OCI Registries?
+
+The Open Container Initiative (OCI) defines the specifications and standards for container technologies. This includes the API for working with container registries, known formally as the OCI Distribution Specification (a.k.a. the "distribution-spec").
+
+The distribution-spec was written based on an open-source registry server originally released by the company Docker, which lives on GitHub at [distribution/distribution](https://github.com/distribution/distribution) (now a CNCF project).
+
+There are now a number of other open-source and commercial distribution-spec implementations, a list of which can be found [here](https://oras.land/oci-registries). Registries that implement the distribution-spec are referred to herein as OCI Registries.
```
Comment (Wwwsylvia) [inline documentation/index.md:17]: The link https://oras.land/oci-registries does not work.

**Review by Wwwsylvia — COMMENTED** [review]
Can we fix the DCO?

**Review by Wwwsylvia — APPROVED** [review]
LGTM
